src/io/json.cpp

```diff
@@ -797,23 +786,9 @@ namespace fc
    }
    variant json::from_file( const fc::path& p, parse_type ptype )
    {
-      //auto tmp = std::make_shared<fc::ifstream>( p, ifstream::binary );
-      //auto tmp = std::make_shared<std::ifstream>( p.generic_string().c_str(), std::ios::binary );
-      //buffered_istream bi( tmp );
-      boost::filesystem::ifstream bi( p, std::ios::binary );
-      switch( ptype )
-      {
-          case legacy_parser:
-              return variant_from_stream<boost::filesystem::ifstream, legacy_parser>( bi );
-          case legacy_parser_with_string_doubles:
-              return variant_from_stream<boost::filesystem::ifstream, legacy_parser_with_string_doubles>( bi );
-          case strict_parser:
-              return json_relaxed::variant_from_stream<boost::filesystem::ifstream, true>( bi );
-          case relaxed_parser:
-              return json_relaxed::variant_from_stream<boost::filesystem::ifstream, false>( bi );
-          default:
-              FC_ASSERT( false, "Unknown JSON parser type {ptype}", ("ptype", ptype) );
-      }
+      fc::istream_ptr in( new fc::ifstream( p ) );
```

Comment:
Need `std::ios::binary` here.

Reply:
By the way, need `std::ios::binary` in `from_file(...)` below as well.

Reply:
ifstream always calls open with std::io::binary, see https://github.com/bitshares/bitshares-fc/blob/c16bb206a39b09b03423088c831bc193de89056c/src/io/fstream.cpp#L64

Reply:
Got it. Thanks. (Just noticed the 2 comments above are talking about same code).